### DISCUSSION TITLE
Does this work for Firebase as a Websocket Client?

### DISCUSSION THREAD

**Discussion by kornha** [darthwalsh/FireSocket#73] [Q&A]
Hello,

I wish to listen to external ws events with firebase backend as the listener (to then publish a pubsub etc),
Will this repo support this? I do not need to run a WSS.

**Comment by darthwalsh** [comment]
Hi @kornha it's been a while since I've looked at this repo and it's in a somewhat abandoned-state (_and I never got it working 100% reliably_).

The goal of this repo is to avoid running a real websocket server: The browser client uses a JS API that has the same methods as [`WebSocket`](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket) but under the hood it's not using websocket communication; instead everything goes through the [Firebase Realtime Database JS SDK](https://firebase.google.com/docs/database/web/start#add_the_js_sdk_and_initialize).

So, if you can ship this JS library to the browser emitting ws events, then yes it supports it!
But if you need to implement the server-side of receiving wb events, then no this isn't the right tool.
